content/en/docs/reference/access-authn-authz/extensible-admission-controllers.md

```diff
@@ -210,4 +197,523 @@ users:
+    // oldObject is the existing object.
+    // It is null for CREATE and CONNECT operations (and for DELETE operations in API servers prior to v1.15.0)
+    "oldObject": {"apiVersion":"autoscaling/v1","kind":"Scale",...},
+    // options contains the options for the operation being admitted, like meta.k8s.io/v1 CreateOptions, UpdateOptions, or DeleteOptions.
+    // It is null for CONNECT operations.
+    // Only sent by v1.15+ API servers.
+    "options": {"apiVersion":"meta.k8s.io/v1","kind":"UpdateOptions",...},
+
+    // dryRun indicates the API request is running in dry run mode and will not be persisted.
+    // Webhooks with side effects should avoid actuating those side effects when dryRun is true.
+    "dryRun": false
+  }
+}
+```
+
+### Response
+
+Webhooks respond with a 200 HTTP status code, `Content-Type: application/json`,
+and a body containing an `AdmissionReview` object (in the same version they were sent),
+with the `response` stanza populated, serialized to JSON.
+
+At a minimum, the `response` stanza must contain the following fields:
+* `uid`, copied from the `request.uid` sent to the webhook
+* `allowed`, either set to `true` or `false`
+
+Example of a minimal response from a webhook to allow a request:
+```json
+{
+  "apiVersion": "admission.k8s.io/v1beta1",
+  "kind": "AdmissionReview",
+  "response": {
+    "uid": "<value from request.uid>",
+    "allowed": true
+  }
+}
+```
+
+Example of a minimal response from a webhook to forbid a request:
+```json
+{
+  "apiVersion": "admission.k8s.io/v1beta1",
+  "kind": "AdmissionReview",
+  "response": {
+    "uid": "<value from request.uid>",
+    "allowed": false
+  }
+}
+```
+
+When rejecting a request, the webhook can customize the http code and message returned to the user using the `status` field.
+The specified status object is returned to the user.
+See [API documentation](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.14/#status-v1-meta) for details about the status type.
+Example of a response to forbid a request, customizing the HTTP status code and message presented to the user:
+```json
+{
+  "apiVersion": "admission.k8s.io/v1beta1",
+  "kind": "AdmissionReview",
+  "response": {
+    "uid": "<value from request.uid>",
+    "allowed": false,
+    "status": {
+      "code": 403,
+      "message": "You cannot do this because it is Tuesday and your name starts with A"
+    }
+  }
+}
+```
+
+When allowing a request, a mutating admission webhook may optionally modify the incoming object as well.
+This is done using the `patch` and `patchType` fields in the response.
+The only currently supported `patchType` is `JSONPatch`.
+See [JSON patch](http://jsonpatch.com/) documentation for more details.
+For `patchType: JSONPatch`, the `patch` field contains a base64-encoded array of JSON patch operations.
+
+As an example, a single patch operation that would add a label would be `[{"op": "add", "path": "/metadata/labels/mylabel", "value": "myvalue"}]`
```

Comment:
Not sure if this should be documented here, but I had a gotcha that when there are no labels `"path": "/metadata/labels/mylabel"` won't work (I would have needed something like`"path": "/metadata/labels", "value": { "mylabel": "myvalue" }` I believe). Naturally this makes me want to use a library to produce a patch for me from a old/new version if an object to avoid having to deal with those types of cases.. Is it worth mentioning that there are utilities to create the patches (or is that obvious)?

Reply:
switched this to setting `spec.replicas`... I don't actually know of tools for generating jsonpatches, and I didn't really want to turn this into a patch doc